projects/observability/src/shared/dashboard/data/graphql/explore/explore-cartesian-data-source.model.ts

```diff
@@ -1,25 +1,98 @@
+import { GraphQlGroupBy } from './../../../../graphql/model/schema/groupby/graphql-group-by';
+import { GraphQlFilter } from '../../../../../../../distributed-tracing/src/shared/graphql/model/schema/filter/graphql-filter';
+import { ColorService, forkJoinSafeEmpty, RequireBy, TimeDuration } from '@hypertrace/common';
+import { GraphQlDataSourceModel, GraphQlSortBySpecification, MetadataService, Specification } from '@hypertrace/distributed-tracing';
+import {
+  ARRAY_PROPERTY,
+  Model,
+  ModelModelPropertyTypeInstance,
+  ModelProperty,
+  ModelPropertyType,
+  NUMBER_PROPERTY,
+  PLAIN_OBJECT_PROPERTY,
+  STRING_PROPERTY
+} from '@hypertrace/hyperdash';
 import { ModelInject } from '@hypertrace/hyperdash-angular';
 import { isEmpty } from 'lodash-es';
-import { EMPTY, Observable } from 'rxjs';
-import { map, switchMap } from 'rxjs/operators';
+import { Observable, of, NEVER } from 'rxjs';
+import { map, mergeMap } from 'rxjs/operators';
 import { Series } from '../../../../components/cartesian/chart';
-import { ExploreVisualizationRequest } from '../../../../components/explore-query-editor/explore-visualization-builder';
+import {
+  ExploreRequestState,
+  ExploreSeries
+} from '../../../../components/explore-query-editor/explore-visualization-builder';
 import { MetricTimeseriesInterval } from '../../../../graphql/model/metric/metric-timeseries';
 import { ExploreSpecification } from '../../../../graphql/model/schema/specifications/explore-specification';
 import {
   ExploreGraphQlQueryHandlerService,
+  EXPLORE_GQL_REQUEST,
+  GraphQlExploreRequest,
   GraphQlExploreResponse
 } from '../../../../graphql/request/handlers/explore/explore-graphql-query-handler.service';
 import { CartesianDataFetcher } from '../../../widgets/charts/cartesian-widget/cartesian-widget.model';
 import { ExploreResult } from './explore-result';
+import { ExploreSelectionSpecificationModel } from '../specifiers/explore-selection-specification.model';
 @Model({
   type: 'explore-cartesian-data-source'
 })
 export class ExploreCartesianDataSourceModel extends GraphQlDataSourceModel<CartesianDataFetcher<ExplorerData>> {
-  public request?: ExploreVisualizationRequest;
+
+  // @ModelProperty({
+  //   key: 'metric',
+  //   // tslint:disable-next-line: no-object-literal-type-assertion
+  //   type: {
+  //     key: ModelPropertyType.TYPE,
+  //     defaultModelClass: ExploreSelectionSpecificationModel
+  //   } as ModelModelPropertyTypeInstance,
+  //   required: true
+  // })
+  // public metricSpecification!: ExploreSelectionSpecificationModel;
+
+  // @ModelProperty({
+  //   key: 'group-by-specifications',
+  //   displayName: 'Group by specifications',
+  //   required: false,
+  //   type: ARRAY_PROPERTY.type
+  // })
+  // public groupBySpecifications: Specification[] = [];
+
+  @ModelProperty({
+    key: 'series',
+    displayName: 'Explore Series',
+    required: false,
+    type: ARRAY_PROPERTY.type
+  })
+  public series: ExploreSeries[] = [];
```

Comment:
Need thoughts on this API. Goal here is to take group by specs and a metric spec as inputs and have a way to build cartesian series configuration that we can use with cartesian widget.

An example usage is Security events cartesian which needs to show three series for each `securityEventScore` (High, medium and low). The current change works for this use case. Problem comes up when we want to assign a specific color to each series. For this to happen, we need a way to specify the series config from the model json.